include/exadg/fluid_structure_interaction/driver.cpp

```diff
@@ -223,17 +223,28 @@ Driver<dim, Number>::coupling_structure_to_ale(VectorType const & displacement_s
 
 template<int dim, typename Number>
 void
-Driver<dim, Number>::coupling_structure_to_fluid(bool const extrapolate) const
+Driver<dim, Number>::coupling_structure_to_fluid(unsigned int iteration) const
 {
   dealii::Timer sub_timer;
   sub_timer.restart();
 
   VectorType velocity_structure;
   structure->pde_operator->initialize_dof_vector(velocity_structure);
-  if(extrapolate)
-    structure->time_integrator->extrapolate_velocity_to_np(velocity_structure);
+  if(iteration == 0)
+  {
+    if(parameters.use_extrapolation)
+    {
+      structure->time_integrator->extrapolate_velocity_to_np(velocity_structure);
+    }
+    else
+    {
+      velocity_structure = structure->time_integrator->get_velocity_n();
+    }
+  }
   else
+  {
     velocity_structure = structure->time_integrator->get_velocity_np();
+  }
```

Comment:
I think we should introduce a ember function here similar to `update_structure_displacement()` introduced in this PR. The logic seems to be the same apart from replacing `displacement` by `velocity`, right?

Reply:
yes! the motivation for the other function was that we use it in multiple places, while a new `update_structure_velocity()` only here. But I guess they follow the same logic, so why not ;)

Reply:
i will come up with a new version this evening, have to teach now ;)